packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1714,7 +1714,10 @@ function emitCredentialParam(
         credential_types.push(type);
       }
     }
-    if (credential_types.length > 0) {
+    if (
+      credential_types.length > 0 &&
+      context.rlcOptions?.addCredentials !== false
```

Comment:
a bug fix for classical layer and api layer doesn't respect the addCredentials: false configuration.